org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/lookup/BlockScope.java

```diff
@@ -99,11 +99,14 @@ protected BlockScope(int kind, Scope parent) {
 /* Create the class scope & binding for the anonymous type.
  */
 public final void addAnonymousType(TypeDeclaration anonymousType, ReferenceBinding superBinding) {
-	ClassScope anonymousClassScope = new ClassScope(this, anonymousType);
-	anonymousClassScope.buildAnonymousTypeBinding(
-		enclosingSourceType(),
-		superBinding);
-
+	// This may have been called from an annotation processor through Elements#getEnumConstantBody()
+	// and binding may have been set. If binding is already set, skip this
+	if (anonymousType.binding == null) {
```

Comment:
So the assumption is, that non-null will only occur if `superBinding` is an enum, right?
Should that be encoded as an assertion?

Reply:
Are you saying if it is non-null and super is an enum, we just don't need the code below and therefore return?